matrix/rules.py

```diff
@@ -376,20 +371,34 @@ def allow_event(e):
                 payload=context.suite)
         for test in context.suite:
             context.test = test
+            success = False
             try:
                 await self.add_model(context)
-                await self.run_once(context, test)
-            finally:
+            except Exception as e:
+                log.error('Error adding model: %s', e)
+                self.exit_code = 200
+            else:
                 try:
-                    await utils.crashdump(
-                        log=log,
-                        tag=context.juju_model.info.name,
-                        directory=context.config.output_dir
-                    )
-                except Exception as e:
-                    log.exception("Error while running crashdump.")
-                if not self.keep_models:
-                    await self.destroy_model(context)
+                    success = await self.run_once(context, test)
+                finally:
+                    try:
+                        await utils.crashdump(
+                            log=log,
+                            tag=context.juju_model.info.name,
+                            directory=context.config.output_dir
+                        )
+                    except Exception as e:
+                        log.exception("Error while running crashdump.")
+                    if not self.keep_models:
+                        try:
+                            await self.destroy_model(context)
+                        except Exception as e:
+                            log.error('Error destroying model: %s', e)
+            log.debug("%s Complete %s %s", test.name, success, context.states)
```

Comment:
This log message will always be excluded from the crash dump, which is a shame, because it tells you whether the test succeeded or failed.